### DISCUSSION TITLE
Unhandled Promise Rejection - Connection Reset

### DISCUSSION THREAD

**Discussion by inbytesdevteam** [vercel/community#156] [Help] [answered]
We're experiencing a 502 Gateway error intermittently on our site for API calls. We currently use Vercel with Nuxt.js and Supabase.

Here's the error it brings out:

```
"stack":["Runtime.UnhandledPromiseRejection: FetchError: request to {API_HERE} failed, reason: Client network socket disconnected before secure TLS connection was established","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:326:22)","    at process.EventEmitter.emit (domain.js:483:12)","    at processEmit [as emit] (/var/task/node_modules/signal-exit/index.js:161:32)","    at processPromiseRejections (internal/process/promises.js:209:33)","    at processTicksAndRejections (internal/process/task_queues.js:98:32)"]
```

I was wondering if anyone has experienced this error before? I have multiple queries in our page.

**Comment by sarukuku** [comment]
This is something I've seen with this service combo (vercel + supabase) too. Curious about the answer.

**Comment by cathykc** [comment]
We're having similar errors with requests to Pusher - would appreciate any leads to investigate :)


```
{
  "errorType":"FetchError",
  "errorMessage":"request to [URL] failed, reason: Client network socket disconnected before secure TLS connection was established",
  "code":"ECONNRESET",
  ...
}
```

**Reply by sarukuku** [reply]
I think we solved our issues related to this and it was just some missing `await`s in the code.

**Reply by TomasHubelbauer** [reply]
Can confirm, in my case I was using API route handler and in it calling the Supabase client like so: supabase.from(table).insert(data).then(console.log) as a quick and dirty way for a PoC. After switching this out for an async/await based approach, it started working.

**Comment by homanp** [comment, marked as answer]
I got the same issue when doing `async/await` inside a `map`. Just wrap the whole thing around a `Promise` and you will mitigate the issue. Makes sense.

**Reply by TanBeige** [reply]
How would I "wrap this whole thing in a promise". I can't seem to find an answer online on how to do that with fetch API calls

`
export async function getServerSideProps(context) {
  try {
    let url = myAPIurl;
    console.log("params:" ,context.params)
    console.log("URL: ", url)

    let resJson;

    const collectionData = await get(url)
    const dataJson = await collectionData.json()
    resJson = dataJson[0];
    console.log(resJson)
  
    return {
      props: {
        collectionRes: resJson,
      }
    };
  }
  catch (err){
    console.log("error", err)
    return {
      props: {
        collectionRes: [null],
      }
    }
  }
}
`
